app/backend/utils.js

```diff
@@ -310,4 +318,85 @@ module.exports.findUserComments = async spec => {
       username: user.name,
       usersurname: user.surname
     }}))
+}
+
+async function resultPredictions() {
+
+  // Get currencies
+  const currencies = await CurrentTradingEquipment.find()
+
+  var currentDay = new Date();
+  day_format = currentDay.toISOString().slice(0,10); // yyyy-mm-dd
+
+  // get today's all prediction data
+  const data = await TradingEquipmentPrediction.find().where('Date').equals(day_format)
+
+  // for each prediction
+  for (const element of data) {
+
+    let TradingEq = element.TradingEq;
+    // get the current value of the predicted currency
+    const currency =  currencies.filter(function(c) {
+      return c.from == TradingEq;
+    });
+
+    let currentValue = currency[0].rate;
+    if(element.Result != "")
+      continue;
+
+    // check whether the value at the prediction time is higher or lower than the current value and determine prediction result accordingly
+    if(element.CurrentValue > currentValue){
```

Comment:
we should discuss about situations when currency value does not change

Reply:
if the value does not change and user predicted either up or down, it eventually predicted wrong. So it is not a true prediction. So we can count this case a wrong prediction.